src/Model/Filter/FilterInterface.php

```diff
@@ -0,0 +1,98 @@
+<?php
+namespace Search\Model\Filter;
+
+use Cake\Datasource\QueryInterface;
+
+/**
+ * Filter Interface
+ */
+interface FilterInterface
+{
+    /**
+     * Get the database field name.
+     *
+     * @return string|array
+     */
+    public function field();
```

Comment:
`field()` can also be dropped and deprecated in `Base`. It's better to just use `fields()` always which has a consistent return type.